src/Compilers/Core/Portable/Operations/IRangeOperation.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    /// <summary>
+    /// Represents a range operation.
+    /// <para>
+    /// Current Usage:
+    ///  (1) C# range expressions
+    /// </para>
+    /// </summary>
+    /// <remarks>
+    /// This interface is reserved for implementation by its associated APIs. We reserve the right to
+    /// change it in the future.
+    /// </remarks>
+    public interface IRangeOperation : IOperation
```

Comment:
>IRangeOperation [](start = 21, length = 15)

I think it would be good to expose a method used to construct the Range instance.